File: src/main/java/com/matag/game/launcher/LauncherTestGameController.java
package com.matag.game.launcher;

import java.util.concurrent.atomic.AtomicLong;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import lombok.AllArgsConstructor;

@Profile("test")
@Controller
@AllArgsConstructor
public class LauncherTestGameController {
  public static final AtomicLong TEST_ADMIN_TOKEN = new AtomicLong();
  private final LauncherGameResponseBuilder launcherGameResponseBuilder;

  @ResponseBody
  @GetMapping(path = {"/ui/test-game"})
  public String launchGame() {
    return launcherGameResponseBuilder.build(String.valueOf(TEST_ADMIN_TOKEN.incrementAndGet()));
  }
}

File: src/main/kotlin/com/matag/game/launcher/LauncherTestGameController.kt
package com.matag.game.launcher

import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ResponseBody
import java.util.concurrent.atomic.AtomicLong

@Profile("test")
@Controller
open class LauncherTestGameController(
    private val launcherGameResponseBuilder: LauncherGameResponseBuilder
) {

    @ResponseBody
    @GetMapping(path = ["/ui/test-game"])
    open fun launchGame(): String {
        return launcherGameResponseBuilder.build(TEST_ADMIN_TOKEN.incrementAndGet().toString())
    }

    companion object {
        val TEST_ADMIN_TOKEN: AtomicLong = AtomicLong()
    }
}
